evalml/objectives/standard_metrics.py

```diff
@@ -813,6 +816,47 @@ def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
         )
 
 
+class MASE(TimeSeriesRegressionObjective):
+    """Mean absolute percentage error for time series regression. Scaled by 100 to return a percentage.
+
+    Only valid for nonzero inputs. Otherwise, will throw a ValueError.
+
+    Example:
+        >>> y_true = pd.Series([1.5, 2, 3, 1, 0.5, 1, 2.5, 2.5, 1, 0.5, 2])
+        >>> y_pred = pd.Series([1.5, 2.5, 2, 1, 0.5, 1, 3, 2.25, 0.75, 0.25, 1.75])
+        >>> np.testing.assert_almost_equal(MAPE().objective_function(y_true, y_pred), 15.9848484)
+    """
+
+    name = "Mean Absolute Scaled Error"
+    greater_is_better = False
+    score_needs_proba = False
+    perfect_score = 0.0
+    is_bounded_like_percentage = False  # Range [0, Inf)
+    expected_range = [0, float("inf")]
+
+    def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
+        """Objective function for mean absolute percentage error for time series regression."""
+        y_train = y_true
+        if (y_train == 0).all():
+            raise ValueError(
+                "Mean Absolute Scaled Error cannot be used when "
+                "all training targets contain the value 0.",
+            )
+        if isinstance(y_true, pd.Series):
+            y_true = y_true.to_numpy()
+        if isinstance(y_predicted, pd.Series):
+            y_predicted = y_predicted.to_numpy()
+        if isinstance(y_train, pd.Series):
+            y_train = y_train.to_numpy()
+        mase = MeanAbsoluteScaledError()
+        return mase(y_true, y_predicted, y_train=y_train) * 100
```

Comment:
According to the documentation, another parameter that we may want to consider leveraging is `sp`. With our `Decomposer.determine_periodicity` function (or, if the pipeline has a Decomposer, the decomposer's saved sp parameter), we should have the information necessary to pass that through. The question remains how, since we need information from X and y train, but it could be done. From brief testing, it looks like adding the sp argument gives us lower values (aka "better" scores), and they would once again be considered to be more accurate, since we use sp during training.

I'm curious what other people think about the need to include sp as well.

Reply:
I agree we should support this! @remyogasawara can you file another issue for this so we can prioritize after decomp is added for multiseries?

Reply:
#4224